apps/coscrad-frontend/src/store/slices/resources/terms/thunks/fetch-terms.ts

```diff
@@ -1,13 +1,144 @@
+ */
+export interface IUserQueryOptions<T> {
+    filter?: IUserDefinedFilter<T>;
+    pagination: {
+        size: number;
+        page: number;
+    };
+}
+
+export const fetchTerms = createAsyncThunk(
+    buildResourceFetchActionPrefix(TERMS),
+    async (options: Partial<IUserQueryOptions<ITermViewModel>> | null, thunkApi) => {
+        const { getState } = thunkApi;
+
+        const token = selectAuthToken(getState() as RootState);
+
+        const optionsWithDefaultsApplied: IUserQueryOptions<ITermViewModel> = {
+            pagination: options?.pagination || {
+                size: 100,
+                page: 1,
+            },
+        };
+
+        if (options?.filter) {
+            optionsWithDefaultsApplied.filter = options.filter;
+        }
+
+        const headers = buildAuthenticationHeaders(token);
+
+        const requestOptions = options
+            ? {
+                  method: 'POST',
+                  headers,
+                  body: JSON.stringify(optionsWithDefaultsApplied),
+              }
+            : {
+                  method: 'POST',
+                  headers,
+              };
+
+        const response = await fetch(`${getApiResourcesBaseRoute()}/terms`, requestOptions);
+
+        const responseJson = await response.json();
+
+        if (response.status !== HttpStatusCode.createdResource)
+            /**
+             * TODO [https://www.pivotaltracker.com/story/show/183619131]
+             *
+             * We need more specific error handling that considers the format of
+             * and difference between a returned error, a system error (backend runtime exception),
+             * and other errors (e.g. not found, not authorized).
+             */
+            return thunkApi.rejectWithValue({
+                code: responseJson.statusCode,
+                message: responseJson.error,
+            } as IHttpErrorInfo);
+
+        const { apiUrl } = getConfig();
+
+        const { indexToDetailFlows } = getConfigurableContent();
+
+        const termIndexToDetailFlowConfig = indexToDetailFlows.find(
+            ({ categorizableType }) => categorizableType === CategorizableType.term
+        );
+
+        const identityFilter = (x: unknown) => x;
+
+        const preFilter = termIndexToDetailFlowConfig?.indexFilter || identityFilter;
+
+        // TODO do this on the back-end
+        const newEntitiesWithAudioUrl = responseJson.entities.flatMap((entity) => {
+            const doesEntityPassFilter = preFilter(entity);
+
+            return doesEntityPassFilter
+                ? [
+                      {
+                          ...entity,
+                          audioURL: isNullOrUndefined(entity.audioURL)
+                              ? undefined
+                              : `${apiUrl}${entity.audioURL}`,
+                      },
+                  ]
+                : [];
+        });
+
+        /**
+         * TODO Phase the following mapping layer out in favour
```

Comment:
Is the back-end already doing this? If not, can we do it now?

Reply:
It looks like the back-end does this already. I'm going to try removing this once I've pushed all other refactors.

Reply:
Let's not forget this.